docs/advanced/embedding.rst

```diff
@@ -105,6 +105,25 @@ The two approaches can also be combined:
         std::cout << message;
     }
 
+Typically, a call to a function in a `py::module` returns a generic `py::object`,
+which can often be autmatically cast to a specialized pybind11 type:
```

Comment:
Calling `py::module::import` returns a `py::module`.

Reply:
That's why I wrote "typically" :wink: 
But maybe that's not true either. Is it?

Reply:
It is true, at least from my reading:
https://github.com/pybind/pybind11/blob/37f845a1dc171d981149a53f01c13c6ba85e0f2e/include/pybind11/pybind11.h#L914

I think the "typically" can be dropped, because in C++, it is `py::module` (the C++ type itself).